From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: hayanesuru <hayanesuru@outlook.jp>
Date: Sat, 28 Feb 2026 14:42:29 +0900
Subject: [PATCH] reduce computeSpeed Vec3 allocations


diff --git a/net/minecraft/world/entity/Entity.java b/net/minecraft/world/entity/Entity.java
index f3399891bc467c792c942ec20c2c920c6bfc205f..103cca2b3440dbe2bbafd7e60cbd35915b4aa773 100644
--- a/net/minecraft/world/entity/Entity.java
+++ b/net/minecraft/world/entity/Entity.java
@@ -1023,7 +1023,7 @@ public abstract class Entity implements SyncedDataHolder, DebugValueSource, Name
             this.lastKnownPosition = this.position();
         }
 
-        this.lastKnownSpeed = this.position().subtract(this.lastKnownPosition);
+        this.lastKnownSpeed = this.position().equals(this.lastKnownPosition) ? Vec3.ZERO : this.position().subtract(this.lastKnownPosition); // Leaf - reduce computeSpeed Vec3 allocations
         this.lastKnownPosition = this.position();
     }
 
