From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Taiyou06 <kaandindar21@gmail.com>
Date: Tue, 26 Nov 2024 19:58:29 -0500
Subject: [PATCH] Remove stream in matchingSlot


diff --git a/src/main/java/net/minecraft/world/item/enchantment/Enchantment.java b/src/main/java/net/minecraft/world/item/enchantment/Enchantment.java
index 7b1c7da451b93ee71db1488bc21fdeaf658da5fc..eb96213c7f1b7880f29fdb6763fad6d2fb7eba62 100644
--- a/src/main/java/net/minecraft/world/item/enchantment/Enchantment.java
+++ b/src/main/java/net/minecraft/world/item/enchantment/Enchantment.java
@@ -163,7 +163,15 @@ public final class Enchantment { // Leaf - Decompile fix
     }
 
     public boolean matchingSlot(EquipmentSlot slot) {
-        return this.definition.slots().stream().anyMatch(slotx -> slotx.test(slot));
+        // Leaf start - Remove stream in matchingSlot
+        for (EquipmentSlotGroup slotx : this.definition.slots()) {
+            if (slotx.test(slot)) {
+                return true;
+            }
+        }
+
+        return false;
+        // Leaf end - Remove stream in matchingSlot
     }
 
     public boolean isPrimaryItem(ItemStack stack) {
